src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1916,8 +1921,19 @@ CreateHashDistributedTableShards(Oid relationId, int shardCount,
 		 * tables which will not be part of an existing colocation group. Therefore,
 		 * we can directly use ShardReplicationFactor global variable here.
 		 */
-		CreateShardsWithRoundRobinPolicy(relationId, shardCount, ShardReplicationFactor,
-										 useExclusiveConnection);
+		if (shardCount == 1 && localTableEmpty && ShardReplicationFactor == 1)
```

Comment:
is the `localTableEmpty` needed? seems a bit unexpected

Reply:
The existing single shard distributed table sets min/max as NULL but a distributed table with single shard still would have  min/max being Long.Min/Long.Max so we can't fully leverage the existing stuff. So had to move this down to the selection logic to handle this when picking the nodes for the shards.